pkg/test/http_recorder.go

```diff
@@ -123,7 +123,7 @@ func (r *HTTPRecorder) record(entry *LogEntry, req *http.Request, resp *http.Res
 		} else if resp.Body != nil {
 			requestBody, err := io.ReadAll(resp.Body)
 			if err != nil {
-				panic(fmt.Sprintf("failed to read response body for request %q: %v", req.URL, err))
+				return fmt.Errorf("failed to read response body for request %q: %w", req.URL, err)
```

Comment:
I am trying to understand what is the implication of this change. If we return an error here, do we ended up having a corrupted http log?

Reply:
Folks, keep me honest, but I don't think so.

AFAICT if we return the err here we don't record these events anymore. 

Taking a step back, we fail to read the body, particularly in this case, because the context was canceled, because the test is not running anymore. So I think it's safe to say we don't care about http events any more since the test is finished.

Reply:
Thank you for help me understand. I will defer to Joyce and Gemma.